pilz_control/include/pilz_control/pilz_joint_trajectory_controller_impl.h

```diff
@@ -95,13 +95,23 @@ template <class SegmentImpl, class HardwareInterface>
 bool PilzJointTrajectoryController<SegmentImpl, HardwareInterface>::
 handleHoldRequest(std_srvs::TriggerRequest&, std_srvs::TriggerResponse& response)
 {
+  if(active_mode_ == Mode::HOLD)
+  {
+    response.message = "Already in hold mode";
+    response.success = true;
+    return true;
+  }
+
   std::lock_guard<std::mutex> lock(sync_mutex_);
+
   JointTrajectoryController::preemptActiveGoal();
   triggerMovementToHoldPosition();
 
   active_update_strategy_ = std::bind(&PilzJointTrajectoryController::updateStrategyWhileHolding, this,
```

Comment:
Since you introduce a state now, you could replace this updateStrategy function pointer with a more readable `if` block inside a single update function. (Get rid of `active_update_strategy_` in favor of using `active_mode_`)

Reply:
Yep :+1: